### ISSUE TITLE
Allow template part queries to have a base identifier and an optional specific id

### ISSUE THREAD

**Issue by centigrade-julian-lang** [Centigrade/ngtx#20]
### Description

Currently the `data-ngtx` attribute can only have a single ID string that can be queried for in tests. Sometimes a user might have several items of the same conceptual-type that they need to query for individually **and** all together.

Example:

~~~html
<nav>
   <a data-ngtx="nav-item">Home</a>
   <a data-ngtx="nav-item">Profile</a>
   <a data-ngtx="nav-item">Logout</a>
</nav>
~~~

Of course we could query for all of them like

~~~ts
getAll('ngtx_nav-item'); // retrieves all three nav-items
~~~

and individually by using the index:

~~~ts
getAll('ngtx_nav-item').nth(1 /*, 2, 3, ... */); // returns only the nth nav-item
~~~

But this would be brittle as the index of the items might change; that's why we use the template parts in the first place - to avoid as many breaking changes as possible for the tests due to template refactorings.

### Proposed solution

A better solution would be to allow a syntax that allows to define a base ID and a specific ID. Something like this:

~~~html
<nav>
   <a data-ngtx="nav-item(home)">Home</a>
   <a data-ngtx="nav-item(profile)">Profile</a>
   <a data-ngtx="nav-item(logout)">Logout</a>
</nav>
~~~

This way we could query for the base ID:

~~~ts
getAll('ngtx_nav-item');
~~~

And for specific IDs as well:

~~~ts
get('ngtx_nav-item(home)'); // returns only the nav-item with the specific ID "home"
~~~

### Alternatives considered

An alternative could be to allow advanced css query selectors, by allowing the template part query to be modified with further css selectors:

~~~html
<a data-ngtx="nav-item" routerLink="home">Home</a>
<!-- ... -->
~~~

And then:

~~~ts
getAll('ngtx_nav-item');
get('ngtx_nav-item [routerLink="home"]');
~~~

But this again would bind us against implementation details and would be brittle as well. Whenever the router link changes, our API would break.

**Comment by centigrade-julian-lang** [comment]
CC @matthiasfrauer @alexanderkeller

**Comment by matthiasfrauer** [comment]
Since we often query for css classes without ngtx, i would propose a css class like solution
```html
<nav>
   <a data-ngtx="nav-item nav-item-home">Home</a>
   <a data-ngtx="nav-item">Profile</a>
   <a data-ngtx="nav-item">Logout</a>
</nav>
```

and access them like this
```js
getAll('ngtx_nav-item');
get('ngtx_nav-item-home');
```

This leverages the users understanding of css class logic and they don't have to learn ngtx specific logic.

**Comment by centigrade-julian-lang** [comment]
Shipped in version `2.1.4-alpha.58`. ngtx queries will now only check via "ngtx attribute starts with the given id", which enables the mentioned behavior out of the box:
~~~html
<nav>
   <a data-ngtx="nav:item.home">Home</a>
   <a data-ngtx="nav:item.profile">Profile</a>
   <a data-ngtx="nav:item.logout">Logout</a>
</nav>
~~~
 ~~~ts
getAll('ngtx_nav:item'); // gets all elements having an "data-ngtx"-attribute starting with "nav:item"
get('ngtx_nav:item.home'); // gets only the home item
~~~

**Comment by centigrade-julian-lang** [comment]
Will add this feature in near future

**Comment by centigrade-julian-lang** [comment]
Shipped in commit a64a245d3c28c8c39daaa4be28ba9674a2149b8e in version `2.4.0`. See docs https://github.com/Centigrade/ngtx/blob/main/docs/querying.md#querying-by-ngtx-id
